bindings/pyroot/pythonizations/python/ROOT/pythonization/_roofit/_utils.py

```diff
@@ -74,6 +81,80 @@ def _string_to_root_attribute(value, lookup_map):
         return value
 
 
+def _dict_to_std_map(arg_dict):
+    """Helper function to convert python dict to std::map."""
+    import ROOT
+
+    allowed_key_types = ["std::string", "RooCategory*"]
```

Comment:
Shouldn't these two lists be a parameter of `_dict_to_std_map`? For example, in the case of `RooCategory`, you only allow `std::map< std::string, int >` IIUC. Depending on the caller, you would pass different lists of allowed keys and values. And perhaps it shouldn't be two lists but a dictionary instead: given a certain key type, the value types can be X and Y (to prevent the user from passing a wrong combination of key-value type).

Reply:
Hi @etejedor, thanks for the comment! Indeed, the allowed types should be passed by the caller, remember @Harshalzzzzzzz we discussed this on mattermost :)

I also like the idea of passing the dictionary with the allowed value types for each key type. But since there are not so many allowed key-value pairs for each dictionary anyway, the code might even be more readable if you just pass a list of allowed template parameters.

So for example, for [RooFit::Import()](https://github.com/root-project/root/blob/master/roofit/roofitcore/inc/RooGlobalFunc.h#L141) the call could look like:
```Python
_dict_to_std_map(args[0], allowed_template_args=["std::string,TH1*", "std::string,RooDataHist*", "std::string, RooDataSet*"])
```
For the other callers, the `allowed_template_args` would be:
* RooCategory : `["std::string, int"]`
* RooDataHist : `["std::string,TH1*", "std::string,RooDataHist*"]`
* Slice : `["RooCategory*,std::string"]`
* Link : `["std::string,RooAbsData*"]`

Of course that also means that the implementation of `_dict_to_std_map` needs to be changed a bit. Does this sound like a good interface to you @etejedor? If you have any question about the implementation @Harshalzzzzzzz, just ping me on mattermost.

Reply:
I think using lists has some repetition that could be avoided (e.g. `std::string` is always the key type for `RooFit::Import`). With a dictionary you iterate first on the keys, find the matching key type, then use the dictionary to find the allowed value types for that key type and iterate on those to find the matching value type, if any. With the lists option you would iterate on the pairs, split the types, compare both for key and value, then pass to the next pair if no match (this potentially has more comparisons for the key type in e.g. in `Roofit:Import`, since you might be comparing the keys to string more than once).

Reply:
Okay that it a good point with the less comparisons, let's go for the dictionary then :+1:

Reply:
@etejedor @guitargeek 
Sir I was thinking if we could pass two lists in _dict_to_std_map rather than a dictionary, something like this :
``` 
_dict_to_std_map(args[0], allowed_key_types=["std::string"],  allowed_value_types=["TH1*","RooDataHist*", "RooDataSet*"])
```
We wouldnt have to check every key and value as we would have to if we pass a dict

Reply:
@Harshalzzzzzzz this could be an alternative but it only works if there's never multiple keys and multiple values (the dictionary provides you that mapping keys to possible values).